sdks/java/harness/src/main/java/org/apache/beam/fn/harness/logging/BeamFnLoggingClient.java

```diff
@@ -142,29 +136,32 @@ public BeamFnLoggingClient(
     inboundObserver = new LogControlObserver();
     logRecordHandler = new LogRecordHandler(options.as(GcsOptions.class).getExecutorService());
     logRecordHandler.setLevel(Level.ALL);
-    outboundObserver = streamObserverFactory.apply(stub::logging, inboundObserver);
+    outboundObserver =
+        (ClientCallStreamObserver<BeamFnApi.LogEntry.List>) stub.logging(inboundObserver);
```

Comment:
Is it essential to the change that this is called earlier, or is that just a simplification?

Reply:
The essential change is getting access to the phaser so it can be terminated, the DirectStreamObserver didn't allow one access and the publishing thread would get stuck waiting on something that would never become ready.